### ISSUE TITLE
Current master will not compile on iojs 2.x

### ISSUE THREAD

**Issue by trygve-lie** [nickdesaulniers/node-nanomsg#89]
iojs 2.0.0 was just released with quite a bit of [large and breaking changes in it](https://github.com/iojs/io.js/blob/master/CHANGELOG.md#2015-05-04-version-200-rvagg). Sadly it seems like master will not compile with this version of iojs:

```sh
  CC(target) Release/obj.target/nanomsg/deps/nanomsg/src/protocols/pubsub/xsub.o
  CC(target) Release/obj.target/nanomsg/deps/nanomsg/src/protocols/reqrep/rep.o
  CC(target) Release/obj.target/nanomsg/deps/nanomsg/src/protocols/reqrep/req.o
make: *** No rule to make target `Release/obj.target/nanomsg/deps/nanomsg/src/protocols/reqrep/task.o', needed by `Release/obj.target/nanomsg.a'.  Stop.
make: Leaving directory `/home/trygve/Dev/node-nanomsg/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1009:12)
gyp ERR! System Linux 3.13.0-51-generic
gyp ERR! command "/usr/local/bin/iojs" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/trygve/Dev/node-nanomsg
gyp ERR! node -v v2.0.0
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok 
```

**Comment by trygve-lie** [comment]
BTW; I've tried to upgrade to latest version of nan. That did not solve it.

**Comment by nickdesaulniers** [comment]
This doesn't look like a compiler error, but an error with the build system.  I wonder if they upgraded the version of gyp, which makes some breaking changes?

**Comment by nickdesaulniers** [comment]
ugh seeing this on travis now: https://travis-ci.org/nickdesaulniers/node-nanomsg/jobs/61343730

**Comment by nickdesaulniers** [comment]
@trygve-lie if you run `node-gyp -h` what's the version used by io.js 2.0?

**Comment by nickdesaulniers** [comment]
also, not sure if running `node-gyp install` is necessary or not.

**Comment by reqshark** [comment]
update nan

**Comment by nickdesaulniers** [comment]
@reqshark , didn't @trygve-lie say he tried?  Can you verify that works?

**Comment by reqshark** [comment]
@nickdesaulniers oh ya he did, ok i'll take a look

**Comment by trygve-lie** [comment]
@nickdesaulniers If I run node-gyp -h it yelds this:

```sh
node-gyp@1.0.3  /usr/local/lib/node_modules/node-gyp
node@2.0.0
```

**Comment by reqshark** [comment]
@trygve-lie, which operating system are you running?

are you still not able to build even after doing this: https://github.com/nickdesaulniers/node-nanomsg/commit/145f7e075f158f77f0a18cafcb3cf41f553d707c?

**Comment by nickdesaulniers** [comment]
@trygve-lie , just published v1.1.1, try upgrading and seeing if the issue is fixed for you.  Feel free to close the issue if it's all good.

**Comment by reqshark** [comment]
thanks for alerting us @trygve-lie! 

if you're unable to compile with the new iojs after the 1.1.1 release, lemme know to reopen
